src/hotspot/share/oops/method.cpp

```diff
@@ -942,12 +942,12 @@ bool Method::is_class_initializer() const {
 
 // A method named <init>, if non-static, is a classic object constructor.
 bool Method::is_object_constructor() const {
-   return name() == vmSymbols::object_initializer_name() && !is_static();
+  return name() == vmSymbols::object_initializer_name() && !is_static();
```

Comment:
Do we still need to check that the method is not static?
Now that constructors and static factories have different names, the check of the static flag looks superfluous (applies to the method below as well, and to the comments above the method declaration).

Reply:
You are correct, any kind of correctness check has already happened during parsing